### PR TITLE
Auto fields

### PR DESCRIPTION
**Pull request by mpessas**
Add support for fields that are "delegated" to the database (e.g. let the database populate the default values). 

See ticket https://code.djangoproject.com/ticket/21454.

### CODE REVIEW

**Comment by timgraham** [comment]
Closing due to lack of follow-up. Please send a new PR if someone can update this, thanks!

**Comment by owais** [comment]
@timgraham I'm resuming work on this. Tracking at #5904

File: django/db/models/sql/compiler.py [hunk 1]
```diff
@@ -957,7 +1007,7 @@ def execute_sql(self, result_type):
             if is_empty:
                 rows = aux_rows
                 is_empty = False
-        return rows
+        return (rows, returning_values)
```
Comment (owais) [inline django/db/models/sql/compiler.py]: @shaib I'm resuming work on this as part of https://github.com/django/django/pull/5904
Could you please give me an example scenario when we could get multiple rows so I can test and fix the case?

Comment (shaib) [inline django/db/models/sql/compiler.py]: Frankly, I don't remember the details here, and I only have time to glimpse the PR now, but it seems that there may be two cases where multiple rows of returned values may be expected: `bulk_create()` on one hand, and an `UPDATE` statement on the other (you wouldn't need to returned values in an ORM `update()` call, but you would need them in the lower-level API).

Comment (owais) [inline django/db/models/sql/compiler.py]: OK. I already have them returning at a lower level but public update method still just returns the number of rows affected. I'll look into the `bulk_create()` case.

Thanks

Comment (owais) [inline django/db/models/sql/compiler.py]: @shaib Looks like bulk_create's public API also does not need this. It does not even update IDs (any auto field) right now. May be we can add support for this later as a new feature?

Comment (timgraham) [inline django/db/models/sql/compiler.py]: Yes, that feature request is tracked in https://code.djangoproject.com/ticket/19527.

**Pull request closed without merging**